### ISSUE TITLE
[BUG] Scaffold rotation && placement checks being broken

### ISSUE THREAD

**Issue by be4dev** [CCBlueX/LiquidBounce#1538]
### LiquidBounce Branch

Nextgen

### LiquidBounce Build/Version

latest

### Operating System

Linux

### Minecraft Version

1.20.1

### Describe the bug

https://www.youtube.com/watch?v=yUk6KL9Eo28
this video shows them
* scaffold hates some spots and ignores them, even though, you can place their by hand
* scaffold love wierd rotatiting (it might be stuck for some time, when clearly you can't place blocks in the air, it also seems to try to place blocks at some illegal places (like it aims at invisible side of the block sometimes)
* sometimes it switches rotation for no reason when bridging
I can don't want to delete all this new fancy code, so @superblaubeere27 pls fix it 

### Steps to reproduce

 - do things I did on the video

### Client Log

```shell
not needed
```

**Comment by superblaubeere27** [comment]
Is this bug still present when AdvantedRotations are disabled?

**Comment by be4dev** [comment]
> Is this bug still present when AdvantedRotations are disabled?

yes, also it much easier to notice weird rotation pattern if you stay at the edge, where you can place blocks, but scaffold can't. It will turn around a lot
